ocs_ci/ocs/cluster.py

```diff
@@ -1209,6 +1209,37 @@ def check_osd_tree_1az_aws(osd_tree, number_of_osds):
     return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
 
 
+def check_ceph_osd_tree():
+    """
+    Checks whether an OSD tree is created/modified correctly.
+    It is a summary of the previous functions: 'check_osd_tree_1az_vmware',
+    'check_osd_tree_3az_aws', 'check_osd_tree_1az_aws'.
+
+    Returns:
+         Boolean: True, if the ceph osd tree is formed correctly. Else False
+
+    """
+    osd_pods = pod.get_osd_pods()
+    # 'ceph osd tree' should show the new osds under right nodes/hosts
+    #  Verification is different for 3 AZ and 1 AZ configs
+    ct_pod = pod.get_ceph_tools_pod()
+    tree_output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph osd tree')
+    if config.ENV_DATA['platform'].lower() == constants.VSPHERE_PLATFORM:
+        return check_osd_tree_1az_vmware(tree_output, len(osd_pods))
+
+    aws_number_of_zones = 3
+    if config.ENV_DATA['platform'].lower() == constants.AWS_PLATFORM:
+        # parse the osd tree. if it contains a node 'rack' then it's a
+        # AWS_1AZ cluster. Else, 3 AWS_3AZ cluster
+        for i in range(len(tree_output['nodes'])):
+            if tree_output['nodes'][i]['name'] in "rack0":
```

Comment:
are there racks in AWS?

Reply:
As far as I see there are racks in AWS. TBH I took the code from https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py#L252. I just export it to an external function.

Reply:
adding @prsurve

Reply:
If the cluster is 3az we dont have a rack label if the cluster is 1 az it will have a rack label 

ip-10-0-131-178.us-east-2.compute.internal   Ready    worker   7h      v1.19.0+b4ffb45   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=i3en.2xlarge,beta.kubernetes.io/os=linux,cluster.ocs.openshift.io/openshift-storage=,failure-domain.beta.kubernetes.io/region=us-east-2,failure-domain.beta.kubernetes.io/zone=us-east-2a,kubernetes.io/arch=amd64,kubernetes.io/hostname=ip-10-0-131-178,kubernetes.io/os=linux,node-role.kubernetes.io/worker=,node.kubernetes.io/instance-type=i3en.2xlarge,node.openshift.io/os_id=rhcos,topology.ebs.csi.aws.com/zone=us-east-2a,topology.kubernetes.io/region=us-east-2,topology.kubernetes.io/zone=us-east-2a